src/BloomTests/XmlHtmlConverterTests.cs

```diff
@@ -132,5 +136,23 @@ public void GetXmlDomFromHtml_HasBandItags_NoExtraNewlinesBefore()
 			var dom = XmlHtmlConverter.GetXmlDomFromHtml(html, false);
 			Assert.That(dom.InnerXml, Is.StringContaining(@"one<b>two</b>three<i>four</i>five <b>six</b>seven <i>eight</i>nine"));
 		}
+
+		[Test]
+		public void GetXmlDomFromHtml_HasEmptyTags_NoRemoveTags()
+		{
+			var html = "<!DOCTYPE html><html><head></head><body><div><u> </u></div></body></html>";
+			var dom = XmlHtmlConverter.GetXmlDomFromHtml(html);
+			var xml = dom.DocumentElement.GetElementsByTagName("body")[0].InnerXml;
+			Assert.AreEqual("<div><u></u></div>", xml);
+		}
+
+		[Test]
+		public void GetXmlDomFromHtml_HasNestedEmptyTags_RemoveSelfClosingTags()
+		{
+			var html = "<!DOCTYPE html><html><head></head><body><div><u><i /></u></div></body></html>";
+			var dom = XmlHtmlConverter.GetXmlDomFromHtml(html);
+			var xml = dom.DocumentElement.GetElementsByTagName("body")[0].InnerXml;
+			Assert.AreEqual("<div><u></u></div>", xml);
```

Comment:
Seems as though, if we remove <u></u> when there's nothing inside it, we should remove it also if that is the outcome of an earlier pass.